sdk/core/azure-core/azure/core/_match_conditions.py

```diff
@@ -30,8 +30,17 @@
 class MatchConditions(Enum):
     """An enum to describe match conditions."""
 
-    Unconditionally = 1  # Matches any condition
-    IfNotModified = 2  # If the target object is not modified. Usually it maps to etag=<specific etag>
-    IfModified = 3  # Only if the target object is modified. Usually it maps to etag!=<specific etag>
-    IfPresent = 4  # If the target object exists. Usually it maps to etag='*'
-    IfMissing = 5  # If the target object does not exist. Usually it maps to etag!='*'
+    #: Matches any condition
+    Unconditionally = 1
+
+    #: If the target object is not modified. Usually it maps to etag=<specific etag>
+    IfNotModified = 2
+
+    #: Only if the target object is modified. Usually it maps to etag!=<specific etag>
+    IfModified = 3
+
+    #: If the target object exists. Usually it maps to etag='*'
+    IfPresent = 4
+
+    #: If the target object does not exist. Usually it maps to etag!='*'
+    IfMissing = 5
```

Comment:
```suggestion
    Unconditionally = 1
    """Matches any condition"""

    IfNotModified = 2
    """If the target object is not modified. Usually it maps to etag=<specific etag>"""

    IfModified = 3
    """Only if the target object is modified. Usually it maps to etag!=<specific etag>"""

    IfPresent = 4
    """If the target object exists. Usually it maps to etag='*'"""

    IfMissing = 5
    """If the target object does not exist. Usually it maps to etag!='*'"""
```

Reply:
As I put in screenshot, it does work with `#:`, and this is how autorest is doing documentation for enum right now. Do you want to change the way autorest generates enum value doc as well?

The screenshot from `#:`
![image](https://user-images.githubusercontent.com/1050156/210465858-a38f9f71-8d54-40f0-881d-746093f4f879.png)

Reply:
After testing, indeed `"""` offers better Intelisence, so I changed them